README.md

```diff
@@ -58,124 +95,177 @@ plugins/gtm-martech/*
-      dataLayer: /* enable/disable the DataLayer initialization (default: enabled) */,
-      dataLayerInstanceName: /* Name of the DataLayer to use for all events. (default 'gtmDataLayer') */,
-      tags: [/* One or more GA4 Measumrent Ids */],
-      containers: {
-        lazy: [/* Zero or more GTM Container Ids to load during Lazy Phase */],
-        lazy: [/* Zero or more GTM Container Ids to load during Delayed Phase */],
-      },
-      pageMetadata: { /* Metadata to pass on during the intializaton of the GA4 tag */ },
-      consent: /* require consent (default: enabled) */,
-      consentCallback: /* Function that handles consent processing, if consent is enabled, this must be specified */,
-      decorateCallback: /* Function to call on each found or loaded Section/Block */,
+### 1. Add Preload Hints
+
+Add the following lines at the end of your `head.html`, to speed up the page load:
+```html
+<link rel="preload" as="script" crossorigin="anonymous" href="/plugins/gtm-martech/src/index.js"/>
+<link rel="preconnect" href="https://www.googletagmanager.com"/>
+```
+
+### 2. Import the Plugin
+
+Import the plugin at the top of your `scripts.js` file:
+```js
+import { GtmMartech } from '../plugins/gtm-martech/src/index.js';
+```
+
+### 3. Initialize the Plugin
+
+Create an instance of the plugin. This should be done near the top of your `scripts.js` file, so that the appropriate methods can be called in their respective lifecycle phase.
+
+```js
+const { eager, lazy, delayed } = new GtmMartech({
+  
+  anaytics: /* enable/disable GA4 (default: enabled) */,
+  dataLayer: /* enable/disable the DataLayer initialization (default: enabled) */,
+  dataLayerInstanceName: /* Name of the DataLayer to use for all events. (default 'gtmDataLayer') */,
+  tags: [/* One or more GA4 Measumrent Ids */],
+  containers: {
+    lazy: [/* Zero or more GTM Container Ids to load during Lazy Phase */],
+    lazy: [/* Zero or more GTM Container Ids to load during Delayed Phase */],
+  },
+  pageMetadata: { /* Metadata to pass on during the intializaton of the GA4 tag */ },
+  consent: /* require consent (default: enabled) */,
+  consentCallback: /* Function that handles consent processing, if consent is enabled, this must be specified */,
+  decorateCallback: /* Function to call on each found or loaded Section/Block */,
+});
+
+```
+Note that:
+- If `consent` is enabled, then by default all [consent types](https://developers.google.com/tag-platform/security/concepts/consent-mode#consent-types) will be set to `denied`. 
+- If `consent` is enabled, then a `consentCallback` must be specified. 
+
+### 4. Call Eager Phase Function
+
+Update the `loadEager` function in your `scripts.js` file, to call plugin's eager phase:
+
+```js
+async function loadEager(doc) {
+  …
+  if (main) {
+    decorateMain(main);
+    await eager();
+    doc.body.classList.add('appear');
+    await loadSection(main.querySelector('.section'), waitForFirstImage);
```

Comment:
I'd be tempted to do:
```suggestion
    doc.body.classList.add('appear');
    await Promise.all([
      eager(),
      loadSection(main.querySelector('.section'), waitForFirstImage),
    ]);
```

that way we load the gtm in parallel while also loading the LCP, instead of being blocking